process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+        filepath = folder + "\\" + filename
+        files_list[project].append(filepath)
+    files_merged = dict(files_list)
+    return files_merged
+
+class Project:
+    """
+    Process project
+    """
+    def __init__(self, project, files):
+        self.project = project
+        self.project_files = files
+    def process_data(self):
+        """
+        This function executes the processing of a project based on the project type e.g. SNP, WES, TSO500 or MokaPipe
+        """
+        if self.project == "WES":
+            Project.process_WES(self)
+        elif self.project == "SNP":
+            Project.process_SNP(self)
+        elif self.project == "TSO500":
+            Project.process_TSO(self)
+        elif self.project == "MokaPipe":
+            Project.process_MokaPipe(self)
+    def process_WES(self):
+        """
+        Process WES Runs
+        """
+        destination_of_files = config.WES_destination_of_files
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            url_wes = get_data(csv_file, '"{}"'.format(destination_of_files))
+            output = download_data(url_wes)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_SNP(self):
+        """
+        Process SNP Runs
+        """
+        destination_of_VCFs = config.SNP_destination_of_VCFs
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            url_snp = get_data(csv_file, '"{}"'.format(destination_of_VCFs))
+            output = download_data(url_snp)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_MokaPipe(self):
+        """
+        Process MokaPipe Runs
+        """
+        # Path to St George's transfer folder
+        StG_transfer = config.StG_transfer
+        for file_path in self.project_files:
+            # asks for final part to the destination folder
+            output_folder = ask_for_folder()
+             # create directory for RPKM and download data
+            path_to_RPKM = output_folder + "\\RPKM"
+            destination_of_Coverage = output_folder + "\\coverage"
+            os.mkdir(path_to_RPKM)
+            os.mkdir(destination_of_Coverage)
+            # import csv file to pandas df
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            # get data frames from the master df
+            df_RPKM = csv_file[csv_file['type'] == 'RPKM']
+            df_coverage = csv_file[csv_file['type'] == 'coverage']
+            df_FHPRS = csv_file[csv_file['type'] == 'FH_PRS']
+            # Create a folder with NGS run name, subfolders cold coverasge and RPKM, download data
+            results = re.search(r"\\(NGS\S+)$", output_folder)

```

Comment:
this is taking everything after NGS to get the run id? note, occasionally we can have NGS500 and NGS501 run together

Reply:
Added a regex recognition. if it fines more than one NGS project in the file name it will break